### DISCUSSION TITLE
Basin Polygon Revisions

### DISCUSSION THREAD

**Discussion by dankovacek** [kratzert/Caravan#20] [General]
Hi, thanks for posting this repo, it's a great low-barrier intro. to GEE for me

I understand a part of the motivation of this repository is to promote contributions of other monitored basins, but I'm wondering if this extends to revisions of the existing set of basins in Caravan?  The Water Survey of Canada has made remarkable progress at reviewing and revising their entire database.   The most recent revision can be accessed [here](https://collaboration.cmc.ec.gc.ca/cmc/hydrometrics/www/HydrometricNetworkBasinPolygons/).

I wanted to see if the updates were substantial, so I first did an inner join of the HYSETS stations and the updated WSC set linked above.  I ran the two sets of about 1200 basins separately through the attribute extraction notebook and plotted the attributes with CoDs < 0.95:

![image](https://user-images.githubusercontent.com/4229284/232181190-20b775f7-31a7-42a8-855b-3ceb2bc6aeef.png)

"Flag" means the polygon was flagged in the HYSETS dataset.  The x-axis represents the HYSETS polygons which I imagine are what's in Caravan, and the y-axis represents the updated WSC polygons for the same basins.

Most I assume are just major corrections, as I've noticed a few polygons are quite different in the updated set.  I'm curious about the slope in the wet `wet_pc_s09`.  i don't see this label with the **s** but I imagine it's still related to area coverage of wetland somehow?

 The other task I'd like to do is figure out how to map the Caravan attributes to those in HYSETS, but that's for another discussion and another day maybe.

Cheers.

**Comment by kratzert** [comment]
Sorry for the late reply, I had and accident which required surgery and wasn't working for the past weeks. This week I am at the EGU and it is still hard for me to write long texts. So if it is okay, I would come back to you next week. Just quick points

- we dont include hysets basins that are flagged. (but there are still significant diffs also for noflag)
- all hydroatlas features consider the entire upstream area but are derived from the subpolygon (hence **s** and not **u**). this is just because of our algorithm which is explained in the paper.
- I would be super interested to see the feature comparison.
- I am not sure how to best update the basin polygons (and hence data) best, but I will talk to some of the co-authors.

**Reply by dankovacek** [reply]
Thanks for your response, and sorry to hear about the injury!  Best wishes for a quick and full recovery.  Hope you're still able to make the most of EGU.

Regarding your third point, below is an attempt to map feature names between the two datasets, at least for the soil (GLHYMPS), land cover (NALCMS-2010) and terrain (EarthEnv DEM90) attribute sources that are cited in Arsenault et al. (2019).  At a first glance, I see only a few attributes that appear directly comparable: (elevation, drainage area, slope, and % cover of forest, crops, and urban.)   Caravan labels on the x-axis, Hysets labels on y.:  

![image](https://user-images.githubusercontent.com/4229284/234721171-aeb7396a-b430-417b-a656-3941a54e2019.png)

The fact that the same polygons produce different precision between attributes suggests to me one or more of:
 * different attribute interpretation (computational method)
 * different source information (Hysets is from NALCMS 2010)
 * computational error somewhere
